src/util/articleRenderers.ts

```diff
@@ -107,6 +106,55 @@ export const renderDesktopArticle = (json: any, articleId: string, articleDetail
     return json // This is HTML probably (the problem is that this is hard to know at this stage, buggy architecture)
 };
 
+export const getSectionsHtml = (sections: any, dump: Dump): string => {
+    let html = '';
+    let prev = 0;
+    let tocLevel = sections[0].toclevel;
+    let sectionsToSplit: any[] = [];
+
+    // Store index of sections on current level.
+    sections.map((section: any, index: number) => {
+        if (tocLevel >= section.toclevel) {
+            tocLevel = section.toclevel;
+            sectionsToSplit.push(index);
+        }
+    });
+
+    // Add first section on current level directly and remove it from list.
+    html += sectionTemplate({
+        section_index: sections[0].toclevel,
+        section_id: sections[0].id,
+        section_anchor: sections[0].anchor,
+        section_line: sections[0].line,
+        section_text: sections[0].text,
+        strings: dump.strings,
+    });
+
+    sectionsToSplit = sectionsToSplit.slice(1);
+
+    // Loop over sections on current level and recursively generate html for it's sub-section.
+    sectionsToSplit.forEach((currentIndex: number) => {
+        html = html.replace(`__SUB_LEVEL_SECTION_${sections[prev].toclevel}__`,
+            prev + 1 === currentIndex ? '' : getSectionsHtml(sections.slice(prev + 1, currentIndex), dump));
```

Comment:
This part should just be integrated directly in the `sectionTemplate()` call to the new paramter `subsections_text`. So no html.replace anymore.